scripts/irods/test/test_resource_types.py

```diff
@@ -2385,6 +2385,153 @@ def test_auto_repl_on__2941(self):
+                os.remove(stale_replica_filename)
+            if os.path.exists(get_filename):
+                os.remove(get_filename)
+
+    def test_direct_request_for_cache_archive_replica(self):
+
+        try:
+            filename = "directrequestcachearchivereplica.txt"
+            get_filename = "getfile.txt"
+            lib.make_file(filename, 10, 'arbitrary')
+
+            logical_path = os.path.join(self.admin.session_collection, filename)
+
+            self.admin.assert_icommand("iput " + filename)
+
+            # attempt to get the replica directly from archive - should fail
+            self.admin.assert_icommand(['iget', '-n1', filename, get_filename], 'STDERR', 'DIRECT_CHILD_ACCESS')
+
+            # attempt to get the replica directly from cache - should pass
+            self.admin.assert_icommand(['iget', '-n0', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+            # get the file without requesting replica or resource
+            self.admin.assert_icommand(['iget', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+            # get the file specifying the compound resource - should pass
+            self.admin.assert_icommand(['iget', '-R', 'demoResc', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+        finally:
+            # cleanup
+            self.admin.assert_icommand("irm -f " + filename)
+            if os.path.exists(filename):
+                os.remove(filename)
+            if os.path.exists(get_filename):
+                os.remove(get_filename)
+
+    def test_direct_request_for_cache_archive_replica_prefer_archive_policy(self):
+
+        try:
+            rule_map = {
+                'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
+                    pep_resource_resolve_hierarchy_pre(*INSTANCE, *CONTEXT, *OUT, *OPERATION, *HOST, *PARSER, *VOTE){
+                        *OUT="compound_resource_cache_refresh_policy=always";
+                 '''),
+                 'irods_rule_engine_plugin-python': textwrap.dedent('''
+                     def pep_resource_resolve_hierarchy_pre(rule_args, callback, rei):
+                         rule_args[2] = 'compound_resource_cache_refresh_policy=always'
+                 ''')
+            }
+
+            # manipulate the core.re to add the new policy
+            with temporary_core_file() as core:
+
+                core.add_rule(rule_map[self.plugin_name])
+
+                # restart the server to reread the new core.re
+                IrodsController().restart()
```

Comment:
Why do we need to restart the server? The core.re file is always refreshed on agent startup.

`test_mode=True` also needs to be passed to `restart()`.

Reply:
Copied from a probably really old test case.  I removed the restart.